### PR TITLE
MC stabilized mode: move from mc_pos_control to mc_att_control

### PR DESCRIPTION
**Pull request by bkueng**
Handling the Stabilized fight mode in mc_pos_control adds unnecessary dependencies. This recently became apparent that Stabilized would not work correctly anymore if there was not position published.
So this PR moves the complete handling into mc_att_control.
Additional benefits:
- ability to stop the position controller completely in Stabilized.
- Easier code path for Stabilized as there are less transformations. I find this important especially since Stabilized is often used as a fail-safe flight mode.

### Testing
- I did extensive testing in SITL and HITL, especially mode changes.
- Flight logs:
  - https://logs.px4.io/plot_app?log=50ef926e-6332-4dc7-aab6-b54a572518cb
  - https://logs.px4.io/plot_app?log=a7ae3482-7174-4ce0-bae4-1180441e6f42
- The code is mostly the same as pre-flight task, so it already comes with further test coverage.

### Additional changes
- Fix for rattitude (fa4f38f58b3cdaaf097f288b335b8e9615dcd419)
- Disable Flight tasks if none of them is supposed to run (Fixes https://github.com/PX4/Firmware/issues/10561)
- Use correct maximum tilt angle parameter for altitude mode
- Restore the pre-flight-task yaw behavior (yaw offset) in Stabilized, which is more intuitive to fly

### CODE REVIEW

File: src/modules/mc_pos_control/mc_pos_control_params.c [hunk 1]
```diff
@@ -334,7 +334,7 @@ PARAM_DEFINE_FLOAT(MPC_TILTMAX_AIR, 45.0f);
  *
  * @unit deg
  * @min 0.0
- * @max 90.0
+ * @max 170.0
```
Comment (MaEtUgR) [inline src/modules/mc_pos_control/mc_pos_control_params.c]: feeling adventurous? 😄 It's a good idea to check corner angles (first in simulation!) to verify the implementation. 180 should theoretically also work or do we run into gimbal lock problems with the setpoint generation?

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 2]
```diff
@@ -619,8 +633,8 @@ MulticopterPositionControl::run()
 		if (_wv_controller != nullptr) {
 
 			// in manual mode we just want to use weathervane if position is controlled as well
-			if (_wv_controller->weathervane_enabled() && !(_control_mode.flag_control_manual_enabled
-					&& !_control_mode.flag_control_position_enabled)) {
+			if (_wv_controller->weathervane_enabled() && (!_control_mode.flag_control_manual_enabled
+					|| _control_mode.flag_control_position_enabled)) {
```
Comment (MaEtUgR) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: !(A && !B)  ->  (!A || B) ✔️
according to de morgan

File: src/modules/mc_pos_control/mc_pos_control_params.c (hunk 1 above)
Comment (bkueng) [reply src/modules/mc_pos_control/mc_pos_control_params.c]: Yes, see the commit message: https://github.com/PX4/Firmware/pull/10805/commits/78da9ff79225f8fce638821e5a359e85b531990c. 180 is fine setpoint-wise, beyond that the yaw inverts. I wanted to account for overshoots, but I realize now that even with overshoots the attitude should be tracked correctly up to 180 degrees setpoints.

File: src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp [hunk 3]
```diff
@@ -249,9 +254,47 @@ void FlightTaskManualAltitude::_respectMaxAltitude()
...
+			// check reset counter and update yaw setpoint if necessary
+			if (_sub_attitude->get().quat_reset_counter != _heading_reset_counter) {
+				_yaw_setpoint += matrix::Eulerf(matrix::Quatf(_sub_attitude->get().delta_q_reset)).psi();
+				_heading_reset_counter = _sub_attitude->get().quat_reset_counter;
+			}
+		}
+	}
+
+	// check if an external yaw handler is active and if yes, let it compute the yaw setpoints
+	if (_ext_yaw_handler != nullptr && _ext_yaw_handler->is_active()) {
```
Comment (RomanBapst) [inline src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp]: @bkueng @MaEtUgR I think we should get rid of this for all non-position controlled flight modes. Weathervane only makes sense if the position is controlled, in fact that's a fundamental assumption of the controller. Also if you want to rename _ext_yaw_handler to something more meaningful I would not be against it (maybe _weathervane_yaw_handler?).

Comment (RomanBapst) [reply src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp]: @MaEtUgR This also answers your question regarding weathervane for stabilized mode.

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 4]
```diff
@@ -1011,6 +1014,8 @@ MulticopterPositionControl::start_flight_task()
 			// No task was activated.
 			_flight_tasks.switchTask(FlightTaskIndex::None);
 		}
+	} else if (should_disable_task) {
```
Comment (RomanBapst) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: @bkueng I was wondering if we could somehow do the same thing for the case when we are a vtol in fixed wing mode. Currently in that case we are just returning at the top of this function. I don't have a good idea though.

**Comment by Avysuarez** [comment]
**Tested on pixhawk 4. Good flight, no issues.**
https://review.px4.io/plot_app?log=9cdf5b96-d563-4163-8dee-9d0066add5e8

File: src/modules/mc_pos_control/mc_pos_control_params.c (hunk 1 above)
Comment (bkueng) [reply src/modules/mc_pos_control/mc_pos_control_params.c]: I tested it again and 180 works fine. Interestingly even above that works now. I increased the limit to 180.

File: src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp (hunk 3 above)
Comment (bkueng) [reply src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp]: I moved it to `FlightTaskManualPosition`. Can you give it a quick test?

File: src/modules/mc_pos_control/mc_pos_control_main.cpp (hunk 4 above)
Comment (bkueng) [reply src/modules/mc_pos_control/mc_pos_control_main.cpp]: I noticed that too. The problem there is that I don't want to continuously switch between a task an None.

**Comment by bkueng** [comment]
All comments addressed.

**Pull request merged**